clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4298,6 +4296,42 @@ void handleFailure(Throwable throwable) {
                         }
                     }
                 }
+
+                @Override
+                boolean handleUnsupportedVersionException(UnsupportedVersionException exception) {
+
+                    // if no max timestamp requests were submitted we should not retry
+                    if (partitionsToQuery.stream()
+                        .flatMap(t -> t.partitions().stream())
+                        .noneMatch(p -> p.timestamp() == ListOffsetsRequest.MAX_TIMESTAMP))
+                        return false;
```

Comment:
As we already iterate over all the partitions below, would it be reasonable to check this there? We could set a boolean flag if there is at least one `MAX_TIMESTAMP`. Also, it would be great if we could add a unit test that covers this case.

Reply:
updated with the flag